src/Processors/Formats/Impl/NativeORCBlockInputFormat.cpp

```diff
@@ -97,6 +97,390 @@ std::unique_ptr<orc::InputStream> asORCInputStreamLoadIntoMemory(ReadBuffer & in
     return std::make_unique<ORCInputStreamFromString>(std::move(file_data), file_size);
 }
 
+static const orc::Type * getORCTypeByName(const orc::Type & schema, const String & name)
+{
+    for (uint64_t i = 0; i != schema.getSubtypeCount(); ++i)
+    {
+        if (boost::iequals(schema.getFieldName(i), name))
+            return schema.getSubtype(i);
+    }
+    return nullptr;
+}
+
+static std::optional<orc::PredicateDataType> convertORCTypeToPredicateType(const orc::Type & type)
+{
+    switch (type.getKind())
+    {
+        case orc::BOOLEAN:
+            return orc::PredicateDataType::BOOLEAN;
+        case orc::BYTE:
+        case orc::SHORT:
+        case orc::INT:
+        case orc::LONG:
+            return orc::PredicateDataType::LONG;
+        case orc::FLOAT:
+        case orc::DOUBLE:
+            return orc::PredicateDataType::FLOAT;
+        case orc::VARCHAR:
```

Comment:
BINARY type should be skipped. Because `orc::PredicateDataType::STRING` requires `stringStatistics` in `ColumnStatistics`. 
Refer to the implementation of `PredicateLeaf::evaluatePredicateMinMax`